config/routes.rb

```diff
@@ -1,3 +1,13 @@
 Rails.application.routes.draw do
+  scope 'api' do
+    resources :users
+
+    scope 'profile' do
+      get '/', to: 'profile#me'
+      put '/', to: 'profile#edit'
+      post 'auth', to: 'profile#auth'
```

Comment:
Good idea to have a `/profile` to get a currently logged user information. 👍 

Btw, just curious: why you decided to make registration and login processes part of the `/profile`? 

In my opinion, profile seems to be focused on user's data and editing it, but not on the authentication and user creation processes. I would probably go with `UserController` handling user registration (creation of a new user) and `AuthenticationController` (or SessionController as we imply user work session, even though we use JWT) handling `login` and `logout` actions.

Anyway, seems to me we have a working token based authentication process now to use later on client side!

Reply:
It is semantically understandable for me:
- get profile data
- edit profile
- register profile 
- login to profile

If it is not so obviously, i can move out auth and register from profile scope